### DISCUSSION TITLE
Current state for

### DISCUSSION THREAD

**Discussion by houtknots** [zachowj/node-red-contrib-home-assistant-websocket#360] [Feature Request]
I think it would be very useful if you could use the _current state_ node to check how long a entitie has been in its current state.

For example the heater has been on for longer than 10 minutes or shorter than 10 minutes.

**Comment by zachowj** [comment]
If you drop a debug node after a current state node and output the entity location, `msg.data` you should see a property `timeSinceChangedMs`. This shows you how long the entity has been in its current state in milliseconds.

**Reply by alfredopironti** [reply]
Maybe you can achieve this with an events:state node using the "for" (timer) constraint?

**Comment by zachowj** [comment]
Added in v0.39.0
